### PR TITLE
Replace Glide with Coil

### PR DESCRIPTION
**Pull request by mdrlzy**

### CODE REVIEW

File: app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt [hunk 1]
```diff
@@ -95,14 +96,15 @@ class PreviewImageViewHolder(
...
-        loadGlideZoomImage(id, path, ivZoom)
+        ImageUtils.loadImage(id, path, ivZoom, limitSize = true)
         loadSubsamplingImage(path, ivSubsampling)
     }
 
     fun reset() = with(binding) {
         progress.isVisible = false
         ivZoom.isVisible = true
         ivZoom.isZoomEnabled = true
+        subsamplingImageLoadFailed = false
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt:107]:
```suggestion
        subsamplingImageLoadSuccess = true
```

File: app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt [hunk 2]
```diff
@@ -113,10 +115,12 @@ class PreviewImageViewHolder(
...
-                if (ivZoom.isZoomed) {
-                    progress.isVisible = true
-                    ivZoom.isZoomEnabled = false
-                    ivZoom.resetZoom()
+                if (!subsamplingImageLoadFailed) {
+                    if (ivZoom.isZoomed) {
+                        progress.isVisible = true
+                        ivZoom.isZoomEnabled = false
+                        ivZoom.resetZoom()
+                    }
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt:123]: We can use positive boolean to improve this block of code. See https://testing.googleblog.com/2023/10/improve-readability-with-positive.html
```suggestion
                if (subsamplingImageLoadSuccess) {
                    with(ivZoom) {
                        if (isZoomed) {
                            isVisible = true
                            isZoomEnabled = false
                            resetZoom()
                        }
                       }
                     }
```

File: app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt [hunk 3]
```diff
@@ -131,7 +135,10 @@ class PreviewImageViewHolder(
                     ivZoom.setImageDrawable(null)
                 }
 
-                override fun onImageLoadError(e: Exception?) {}
+                override fun onImageLoadError(e: Exception?) {
+                    subsamplingImageLoadFailed = true
```
Comment (hieuwu) [inline app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt:139]:
```suggestion
                    subsamplingImageLoadSuccess = false
```

File: app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt (hunk 2 above)
Comment (mdrlzy) [reply app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt:123]: This condition should only be met until subsampling image fails. We cannot claim that subsampling image was successfully loaded from very beginning. It hasn't failed or been successful yet. But I'll add comments to make it clear

Comment (hieuwu) [reply app/src/main/java/dev/arkbuilders/navigator/presentation/screen/gallery/previewpager/PreviewImageViewHolder.kt:123]: Okay that makes sense

**Review by hieuwu — DISMISSED** [review]

**Comment by kirillt** [comment]
@mdrlzy the PR in the arklib-android has been merged:
- https://github.com/ARK-Builders/arklib-android/pull/115

**Comment by kirillt** [comment]
@mdrlzy any chance to complete this one? :wink:

**Comment by mdrlzy** [comment]
> @mdrlzy the PR in the arklib-android has been merged:

Yeah, but there is no new release of `arklib-android` that includes these changes)

**Comment by kirillt** [comment]
@mdrlzy thanks for letting know, I've published `0.3.4` now:
https://github.com/ARK-Builders/arklib-android/packages/1935613

**Comment by mdrlzy** [comment]
Test files
[Test_GifSvg.zip](https://github.com/ARK-Builders/ARK-Navigator/files/13810012/Test_GifSvg.zip)

**Review by kirillt — APPROVED** [review]
Awesome, thanks!

**Pull request merged**